src/Databases/PostgreSQL/fetchPostgreSQLTableStructure.cpp

```diff
@@ -54,19 +54,30 @@ static DataTypePtr convertPostgreSQLDataType(std::string & type, bool is_nullabl
         res = std::make_shared<DataTypeDate>();
     else if (type.starts_with("numeric"))
     {
-        /// Numeric and decimal will both end up here as numeric.
-        res = DataTypeFactory::instance().get(type);
-        uint32_t precision = getDecimalPrecision(*res);
-        uint32_t scale = getDecimalScale(*res);
-
-        if (precision <= DecimalUtils::maxPrecision<Decimal32>())
-            res = std::make_shared<DataTypeDecimal<Decimal32>>(precision, scale);
-        else if (precision <= DecimalUtils::maxPrecision<Decimal64>())
-            res = std::make_shared<DataTypeDecimal<Decimal64>>(precision, scale);
-        else if (precision <= DecimalUtils::maxPrecision<Decimal128>())
-            res = std::make_shared<DataTypeDecimal<Decimal128>>(precision, scale);
-        else if (precision <= DecimalUtils::maxPrecision<Decimal256>())
-            res = std::make_shared<DataTypeDecimal<Decimal256>>(precision, scale);
+        /// Numeric and decimal will both end up here as numeric. If it has type and precision,
+        /// there will be Numeric(x, y), otherwise just Numeric
+        uint32_t precision, scale;
+        if (type.ends_with(")"))
+        {
+            res = DataTypeFactory::instance().get(type);
+            precision = getDecimalPrecision(*res);
+            scale = getDecimalScale(*res);
+
+            if (precision <= DecimalUtils::maxPrecision<Decimal32>())
+                res = std::make_shared<DataTypeDecimal<Decimal32>>(precision, scale);
+            else if (precision <= DecimalUtils::maxPrecision<Decimal64>())
+                res = std::make_shared<DataTypeDecimal<Decimal64>>(precision, scale);
+            else if (precision <= DecimalUtils::maxPrecision<Decimal128>())
+                res = std::make_shared<DataTypeDecimal<Decimal128>>(precision, scale);
+            else if (precision <= DecimalUtils::maxPrecision<Decimal256>())
+                res = std::make_shared<DataTypeDecimal<Decimal256>>(precision, scale);
+        }
+        else
+        {
+            precision = DecimalUtils::maxPrecision<Decimal128>();
+            res = std::make_shared<DataTypeDecimal<Decimal128>>(precision, precision);
```

Comment:
Why Decimal128, probably we should use Decimal256 ? And is that correct that for numeric in postgres precision and scale are equal ?
It seems that from https://www.postgresql.org/docs/9.1/datatype-numeric.html#DATATYPE-NUMERIC-TABLE.
```
NUMERIC
without any precision or scale creates a column in which numeric values of any precision and scale can be stored, up to the implementation limit on precision. 
```

For example
```
38 = max precision for Decimal 128.
CREATE TABLE test_decimal (value Decimal(38, 38)) ENGINE=TinyLog
INSERT INTO test_decimal VALUES (2.0);
Code: 69. DB::Exception: Decimal value is too big: 1 digits were read: 2e0. Expected to read decimal with scale 38 and precision 38:  at row 0: data for INSERT was parsed from query
```

From my perspective according to postgreSQL documentation the best we can do is to use Decimal256 and set scale and precision both to 38. What do you think ?

Reply:
May be 76 instead of 38?

Reply:
@kssenii I am not sure but it seems that scale and precision in sum for Decimal256 must be 76 ?

Reply:
@kssenii misread 76 for precision 38 for scale.

Reply:
If referring to doc, they can be equal. 
>  P - precision. Valid range: [ 1 : 76 ]. Determines how many decimal digits number can have (including fraction).
> S - scale. Valid range: [ 0 : P ]. Determines how many decimal digits fraction can have.

Reply:
@kssenii  If they will be equal we could not insert value greater than zero right ?

Reply:
@kssenii greater than one.

Reply:
>  the best we can do is to use Decimal256 and set scale and precision both to 38

But it with this it will be the same then. Seems like 76 precision and 38 scale is ok?

Reply:
Yes Decimal256, 76 precision, 38 scale will be best.

Reply:
Lets leave it Decimal128 instead of 256: `because experimental bigint types are not allowed. Set 'allow_experimental_bigint_types' setting to enable..`